doc/nrf/ug_multi_image.rst

```diff
@@ -24,12 +42,143 @@ MCUboot bootloader
+In the |NCS|, the code is included in the :file:`CMakeLists.txt` file for the samples, and in the MCUboot repository.
+
+See the following example code:
+
+.. code-block:: cmake
+
+   if (CONFIG_SPM)
+     add_child_image(spm ${CMAKE_CURRENT_LIST_DIR}/nrf9160/spm)
+   endif()
+
+   if (CONFIG_SECURE_BOOT)
+     add_child_image(b0 ${CMAKE_CURRENT_LIST_DIR}/bootloader)
+   endif()
+
+   if (CONFIG_BOOTLOADER_MCUBOOT)
+     add_child_image(mcuboot ${MCUBOOT_BASE}/boot/zephyr)
+   endif()
+
+In this code, ``add_child_image`` registers the child image with the given name and file path and executes the build scripts of the child image.
+Note that both the child image's application build scripts and the core build scripts are executed.
+The core build scripts might use a different configuration and possibly different DeviceTree settings.
+
+Adding configuration options
+============================
+
+When enabling a child image, you select the build strategy, thus how the image is included.
+The three options are:
+
+* Build the child image from source along with the parent image - *IMAGENAME*\_BUILD_STRATEGY_FROM_SOURCE
+* Merge the specified HEX file of the child image with the parent image - *IMAGENAME*\_BUILD_STRATEGY_USE_HEX_FILE, and *IMAGENAME*\_HEX_FILE to specify the HEX file
+* Ignore the child image when building and build only the parent image - *IMAGENAME*\_BUILD_STRATEGY_SKIP_BUILD
+
+You must add these four configuration options to the Kconfig file for your child image, replacing *IMAGENAME* with the (uppercase) name of your child image (as specified in ``add_child_image``).
+
+The following example shows the configuration options for MCUboot:
+
+.. code-block:: Kconfig
+
+   choice
+  	prompt "MCUboot build strategy"
+  	default MCUBOOT_BUILD_STRATEGY_FROM_SOURCE
+
+   config MCUBOOT_BUILD_STRATEGY_USE_HEX_FILE
+  	# Mandatory option when being built through 'add_child_image'
+  	bool "Use HEX file instead of building MCUboot"
+
+   if MCUBOOT_BUILD_STRATEGY_USE_HEX_FILE
+
+   config MCUBOOT_HEX_FILE
+  	# Mandatory option when being built through 'add_child_image'
+  	string "MCUboot HEX file"
+
+   endif # MCUBOOT_USE_HEX_FILE
+
+   config MCUBOOT_BUILD_STRATEGY_SKIP_BUILD
+  	# Mandatory option when being built through 'add_child_image'
+  	bool "Skip building MCUboot"
+
+   config MCUBOOT_BUILD_STRATEGY_FROM_SOURCE
+  	# Mandatory option when being built through 'add_child_image'
+  	bool "Build from source"
+
+   endchoice
+
+
+Image-specific variables
+========================
+
+The child image and parent image are executed in different CMake processes and thus have different namespaces.
+Variables in the parent image are not propagated to the child image, with the following exceptions:
+
+* Any variable named *IMAGENAME*\_FOO in a parent image is propagated to the child image named *IMAGENAME* as FOO.
+* Variables that are in the list ``SHARED_MULTI_IMAGE_VARIABLES`` are propagated to all child images.
```

Comment:
Where and how would you define that list?

Reply:
It's a dynamic list, it could be amended from anywhere before add_child_image is invoked, including from CLI.

Currently only the variable EXTRA_KCONFIG_TARGETS and some variables related to it are in the list, but these variables are not interesting for 99% of users.

Not sure if/how to document this.

It is expected that eventually there will be more variables in this list, perhaps variables that users care about, at which point we should definitely list them here.

Reply:
So basically, users would just define that list somewhere in their cmake file - correct?
That's what I would have guessed as well, so maybe we don't need to explain it in more detail.

Reply:
Corret